src/cli/setup_mode.rs

```diff
@@ -115,10 +115,17 @@ pub async fn main(
         warn!("{}", common::WARN_COMPLETE_PROFILE);
     }
 
-    if quiet && process.var("RUSTUP_LOG").is_err() {
-        console_filter
-            .modify(|it| *it = EnvFilter::new("rustup=WARN"))
-            .expect("error reloading `EnvFilter` for console_logger");
+    if process.var("RUSTUP_LOG").is_err() {
+        if quiet {
+            console_filter
```

Comment:
Wonder whether we could deduplicate this a bit more.

Reply:
@djc Done.

Reply:
I would suggest like this:

```rust
let filter = match (quiet, verbose) {
    (true, _) => Some("rustup=WARN"),
    (_, true) => Some("rustup=DEBUG"),
    (_, _) => None,
};

if let Some(filter) = filter {
    console_filter.modify(|it| *it = EnvFilter::new(filter)).expect(..);
}
```

Reply:
@djc Sure thing!